cmd/bundle/exec.go

```diff
@@ -0,0 +1,152 @@
+package bundle
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/databricks/cli/bundle/config/mutator"
+	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/libs/auth"
+	"github.com/spf13/cobra"
+)
+
+func newExecCommand() *cobra.Command {
+	execCmd := &cobra.Command{
+		Use:   "exec",
+		Short: "Execute a command using the same authentication context as the bundle",
+		Args:  cobra.MinimumNArgs(1),
+		Long: `Execute a command using the same authentication context as the bundle
+
+The current working directory of the provided command will be set to the root
+of the bundle.
+
+Example usage:
+1. databricks bundle exec -- echo "hello, world"
+2. databricks bundle exec -- /bin/bash -c "echo hello"
+3. databricks bundle exec -- uv run pytest`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.ArgsLenAtDash() != 0 {
+				return fmt.Errorf("Please add a '--' separator. Usage: 'databricks bundle exec -- %s'", strings.Join(args, " "))
+			}
+
+			// Load the bundle configuration to get the authentication credentials.
+			b, diags := root.MustConfigureBundle(cmd)
+			if diags.HasError() {
+				return diags.Error()
+			}
+
+			childCmd := exec.Command(args[0], args[1:]...)
+
+			env := auth.ProcessEnv(root.ConfigUsed(cmd.Context()))
+
+			// If user has specified a target, pass it to the child command. DABs
+			// defines a "default" target which is a placeholder for when no target is defined.
+			// If that's the case, i.e. no targets are defined, then do not pass the target.
+			//
+			// This is only useful for when the Databricks CLI is the child command.
+			if b.Config.Bundle.Target != mutator.DefaultTargetPlaceholder {
```

Comment:
Technically a bundle can have targets named "default" and "other" and none of them set to `default: true`. Which means user will pass `default` explicitly. Shall we just check for target flag here instead?

Reply:
Good point! Thanks, updated the code to read directly from the flag / env var here.